specs/eslint-configs/presets.test.mts

```diff
@@ -14,28 +15,48 @@ const sampleFiles = (await readdir(srcDir)).map((filename) =>
   resolve(srcDir, filename),
 );
 
-const parsedConfigs = await Promise.all(
-  sampleFiles.map(async (file) => [
-    basename(file),
-    await new ESLint({
-      cwd: rootDir,
-    }).calculateConfigForFile(file),
-  ]),
-);
+const getParsedConfigByOptions = async (
```

Comment:
We should include explanatory comments here detailing why the ESLint API cannot be directly utilized for this test case.